doc/design/new-user-questionnaire.md

```diff
@@ -0,0 +1,56 @@
+- User avatar circley thing needs to show user's current status
+
+### New views
+- SSR rendered pages for questionnaire form, pending approval page, rejected page, and admin approval page.
+
+## 2. High-level Overview
+### Current user signup flow
+1) The user navigates to https://gui.dandiarchive.org and clicks the "LOG IN WITH GITHUB" button
+2) The user is redirected to GitHub, enters their credentials, and clicks "Sign in". This action creates a new DANDI Archive account for the user and gives
+them full capabilities of any other normal user (creating/editing dandisets, etc).
+3) The user is redirected back to https://gui.dandiarchive.org signed in to their new account.
+
+### Proposed user signup flow
+1) The user navigates to https://gui.dandiarchive.org and clicks the "LOG IN WITH GITHUB" button.
+2) The user is redirected to GitHub, enters their credentials, and clicks "Sign in". This action creates a new DANDI archive account for the user with the `status` field set to `INCOMPLETE`.
+3) The user is redirected to the SSR rendered questionnaire form and they fill it out and submit it. This populates the `questionnaire_form` field in the DB and sets the `status` field to `PENDING`.
+4) A message is displayed telling the user when they can expect approval.
+5) The user is redirected to the DANDI Archive homepage. The "LOG IN WITH GITHUB" button is now red and says something like "PENDING ACCOUNT APPROVAL". If the
+user hasn't filled out the questionnaire yet, the button will say "COMPLETE SIGNUP PROCESS" and will redirect the user to the questionnaire form when clicked.
+6) The DANDI admins will decide whether to approve or reject the user.
+7) a) If approved, the user will receive an email saying so and they will be able to logon and use DANDI as normal.
+   <br>
+   b) If rejected, the user will receive an email containing the custom reason (if one was provided), as well as an email address to contact if they belieave
+   they shouldn't have been rejected.
+
+
+### Proposed admin approval process
+1) A user signs up as described in steps 1-4 in the previous section.
+2) An email is sent to DANDI admins with a link to an SSR page that lists the user's info (name, email, GitHub account, etc) as well as their answers to the questionnaire. The page will contain two buttons - "APPROVE" or "REJECT" - and a textbox underneath the "REJECT" button entitled "Rejection Reason".
+3)
+    a) The admin decides to approve the user and clicks "APPROVE". This updates the user's `status` field to `APPROVED` and adds them to the `approved_users` group, granting them full access to DANDI.
+    <br>
+    b) The admin decides to reject the user, optionally fills out the "Rejection reason" box, and clicks "REJECT". This sets the user's `status` field to `REJECTED` and `is_active` field to `false`.
+
+### Questionnaire
+(from https://github.com/dandi/dandi-api/pull/454#issuecomment-892218858)
+- What do you plan to use DANDI for?
```

Comment:
could be a free form text and/or (?) list of multiselect checkboxes

- [ ] upload data for publication
- [ ] collaboration
- [ ] DANDI Hub for computing

Reply:
Not sure if we should wake up a bear of GDPR here with a question either user is "from EU".

That questionnaire form should probably have a URL to our "Terms of Service" to which user should click "Agree" as well (not sure if then we should record here the version of the current Terms of Service since people might agree to one version and later we would issue a new one and might need to alert old users to its availability... brings point also, if possible, to include "or later version" in our Terms)

Reply:
also I wonder if among those 3 options I listed, we could add `download data`, and if only that one is selected, we could alert user that they do not even need to register to get access to the data? (although might be too late since we already added a record for them, right?)

Reply:
> Not sure if we should wake up a bear of GDPR here with a question either user is "from EU".
> 
> That questionnaire form should probably have a URL to our "Terms of Service" to which user should click "Agree" as well (not sure if then we should record here the version of the current Terms of Service since people might agree to one version and later we would issue a new one and might need to alert old users to its availability... brings point also, if possible, to include "or later version" in our Terms)

Let's defer this. We are not lawyers, so binding people at signup time with a ToS doesn't seem like a good idea, especially if we don't know how to keep the door open for future updates.

On that particular topic, my own experience (which must be the same as everyone else's) is that services update their TsoS all the time, and the standard practice seems to be that if I don't like the new terms, I can delete my account, etc. I assume that's how we would handle it ourselves. (i.e., no need to include language that covers changes to the TsoS in the original ToS.)

Reply:
> also I wonder if among those 3 options I listed, we could add `download data`, and if only that one is selected, we could alert user that they do not even need to register to get access to the data? (although might be too late since we already added a record for them, right?)

I don't think it's worthwhile for us to *discourage* people from making an account, even if they only want to download data.

It might be prudent to let people know up front that they can browse and download without an account, but yes by the time they agree to create an account, I think we should commit and not worry about this case where they just need to download things.

Reply:
Thinking about it again, I don't think a Terms of Service should appear alongside the questionnaire. If anything, a ToS should appear as soon as a user wishes to create an account, and the process should abort if they don't accept. But that's not something we're tackling in this workflow, so I'd like to open a new issue to discuss how a ToS can be presented to the user.

Reply:
ok, let's place ToS aside since we don't have any formulated yet.  But we should really get to it and indeed agreeing to ToS should be done while initiating account creation (well -- this questionnair is part of the process)